### DISCUSSION TITLE
How to clear the value of a TextField in Flet?

### DISCUSSION THREAD

**Discussion by AnshThaker** [flet-dev/flet#2551] [Q&A] [answered]
There isn't an obvious way to clear the value of a TextField in Flet, as far as I'm aware of.
Doing this didn't work:
`textfield.value = ''`
So, how do you clear the value of a TextField?

**Comment by ndonkoHenri** [comment, marked as answer]
After doing that, you need to call an update by using either `textfield.update()` or `page.update()`.
If you still face issues after trying please share some repro code so we can help.

**Reply by Vallekralle** [reply]
First of all ndonkoHenri is 100% right in his comment. I just wanted to add: If you do it in a seperate method, where you don't have the capability to use the variable of the object, you have to use the ControlEvent for that, like:

### Example

```
# There should be a “self” in front of the e: ControlEvent, if it was in a class, like ...(self, e: ControlEvent)...
def method_name(e: ControlEvent) -> None:
    e.control.value = ""
    e.control.update()
```

Stay safe and have fun programming!

**Reply by AnshThaker** [reply]
Thanks! Next time I'll make sure I don't forget to update the page 😅
